File: golang-patterns/ObserverPattern/publisher/mensaje.go
package publisher

import "ObserverPattern/subscriber"

type Publisher interface {
	AddSubscriber(n string, s subscriber.Subscriber)
	RemoveSubscriber(n string)
	Notificar(mj string)
}
type mensaje struct {
	msg string
	subscribers map[string]subscriber.Subscriber
}

func (m *mensaje) AddSubscriber(n string, s subscriber.Subscriber) {
	if m.subscribers==nil{
		m.subscribers=make(map[string]subscriber.Subscriber)
	}
	m.subscribers[n]=s
}

func (m *mensaje) RemoveSubscriber(n string)  {
	delete(m.subscribers, n)
}

func (m *mensaje) Notificar(mj string)  {
	m.msg=mj
	for _,s := range m.subscribers{
		s.Update(mj)
	}
}

func NewMensaje() Publisher{
	return &mensaje{}
}

File: java-patterns/observer/src/Mensaje.java
import java.util.ArrayList;

public class Mensaje implements Publisher{
    private String msg;
    private ArrayList<Subscriber> subscribers = new ArrayList<>();

    @Override
    public void addSubscriber(Subscriber s) {
        this.subscribers.add(s);
    }

    @Override
    public void removeSubscriber(Subscriber s) {
        this.subscribers.remove(s);
    }

    @Override
    public void notificar(String event) {
        this.msg=event;
        for (Subscriber s : subscribers){
            s.update(this.msg);
        }
    }
}
